manifests/client.pp

```diff
@@ -12,10 +12,10 @@
 # @param package_ensure 
 #   Ensure the client package is installed
 class postgresql::client (
-  Enum['file', 'absent'] $file_ensure        = 'file',
-  Stdlib::Absolutepath $validcon_script_path = $postgresql::params::validcon_script_path,
-  String[1] $package_name                    = $postgresql::params::client_package_name,
-  String[1] $package_ensure                  = 'present'
+  Enum['file', 'absent'] $file_ensure          = 'file',
+  Stdlib::Absolutepath   $validcon_script_path = $postgresql::params::validcon_script_path,
+  String[1]              $package_name         = $postgresql::params::client_package_name,
+  Variant[Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'], Pattern[/./]] $package_ensure = 'present'
```

Comment:
It is indeed Pattern[/./]. The reason was commented on another review above. Long story short, its mainly for documentation purposes.

Reply:
I get the documentation purposes, but I still think `String[1]` is easier to understand than `Pattern[/./]`, mostly because the . is misleading. `String[1]` parses easily in my mind to "non-empty string". Regexes are harder.

Reply:
I would recomment to go with Stdlib::Ensure::Package then. Everybody who is interested can follow this. Also possible values can be mentioned in the parameter docs.

Reply:
Brought this up as a discussion and I think we can go ahead with the:

Variant[Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'], String[1]] $package_ensure = 'present',

suggestion. It fairly easy to read and documentation should remain comprehensive enough.